stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -833,6 +833,275 @@ opnorm(v::AdjointAbsVec, q::Real) = q == Inf ? norm(conj(v.parent), 1) : norm(co
 opnorm(v::AdjointAbsVec) = norm(conj(v.parent))
 opnorm(v::TransposeAbsVec) = norm(v.parent)
 
+_isparallel(x, y) = abs(dot(x, y)) == length(x)
+
+function _any_cols_are_parallel(X, xrange, Y, yrange)
+    n = size(X, 1)
+    for i in xrange
+        x = view(X,1:n,i)
+        for j in yrange
+            y = view(Y,1:n,j)
+            _isparallel(x, y) && return true
+        end
+    end
+    return false
+end
+
+function _each_col_has_parallel_col(X, Y)
+    for x in eachcol(X)
+        any(y -> _isparallel(x, y), eachcol(Y)) || return false
+    end
+    return true
+end
+
+"""
+    opnormest1(A, t::Integer = 2) -> est
+
+Estimate the operator 1-norm [`opnorm(A, 1)`](@ref) of the matrix or linear operator `A`
+using a block algorithm.
+
+The estimate is a lower bound on the 1-norm and can be computed much more efficiently for
+large and sparse matrices (see Algorithm 2.4 of [^HighamTisseur]). `t` is a parameter that
+sets the size of an ``n × t`` matrix that is multiplied by `A` and must be less than `n`.
+
+[^HighamTisseur]:
+    Nicholas J. Higham and Françoise Tisseur,
+    "A block algorithm for matrix 1-norm estimation, with an application to 1-norm pseudospectra",
+    SIAM Journal on Matrix Analysis and Applications, 21(4), 2000, pp. 1185-1201.
+    [doi:10.1137/S0895479899356080](https://doi.org/10.1137/S0895479899356080)
+
+`A` can be of any type `Op`, representing a square matrix, that implements the following
+methods:
+- `size(A::Op)`
+- `eltype(A::Op)`
+- `*(A::Op, B::AbstractMatrix)`
+- `adjoint(A::Op)`
+
+Note: the algorithm uses random numbers, so the result may change between evaluations.
+
+    opnormest1(A, t::Integer, retv::Val{true}) -> (est, v)
+    opnormest1(A, t::Integer, retv::Val{false}, retw::Val{true}) -> (est, w)
+    opnormest1(A, t::Integer, retv::Val{true}, retw::Val{true}) -> (est, v, w)
+
+Along with the estimate of the operator 1-norm, return a vector `v` and/or a vector `w` that
+correspond to the estimate, such that ``w = A v`` and
+``\\|w\\|_1 = \\mathrm{est} \\|v\\|_1``, where ``\\mathrm{est}`` is the estimate of the
+matrix 1-norm of `A`, and ``\\|⋅\\|`` is the Frobenius 1-[`norm`](@ref).
+"""
+function opnormest1(
+    A,
+    t::Integer=min(2,maximum(size(A))),
+    ::Val{retv}=Val(false),
+    ::Val{retw}=Val(false),
+) where {retv,retw}
+    T = eltype(A)
+    maxiter = 5
+    # Check the input
+    n = checksquare(A)
```

Comment:
Currently limited to square matrices. We can extend to non-square by implicitly padding with zeros, e.g. by wrapping in a `SquareMat`  linear operator (similar to `InvMat`) (recommended in https://www.maths.manchester.ac.uk/~higham/narep/narep135.pdf).

Reply:
It was pretty easy to tweak the algorithm to implicitly pad with zeros.